tuf/repository_lib.py

```diff
@@ -1412,6 +1395,23 @@ def generate_targets_metadata(targets_directory, target_files, version,
 
   if delegations is not None:
     tuf.formats.DELEGATIONS_SCHEMA.check_match(delegations)
+    # If targets role has delegations, collect the up-to-date 'keyids' and
+    # 'threshold' for each role. Update the delegations keys dictionary.
+    delegations_keys = []
+    # Update 'keyids' and 'threshold' for each delegated role
+    for role in delegations['roles']:
+      role['keyids'] = \
+          tuf.roledb.get_role_keyids(role['name'], repository_name)
+      role['threshold']= \
+          tuf.roledb.get_role_threshold(role['name'], repository_name)
```

Comment:
The [coding style](https://github.com/secure-systems-lab/code-style-guidelines#code-lay-out) suggests:
> The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. If necessary, you can add an extra pair of parentheses around an expression, but sometimes using a backslash looks better.

Which suits my personal preference too.
```suggestion
      role['keyids'] = tuf.roledb.get_role_keyids(role['name'],
          repository_name)
      role['threshold']= tuf.roledb.get_role_threshold(role['name'],
          repository_name)
```